packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -62,27 +54,6 @@ export function objectToAST(
     );
 }
 
-export function containsDynamicChildren(parent: ParentNode): boolean {
```

Comment:
I don't see any base case that can return `true`, so right now we're doing a lot of traversal just to return `false` in the end. Am I missing something?